core/google/cloud/credentials.py

```diff
@@ -106,15 +108,15 @@ def _get_signed_query_params(credentials, expiration, string_to_sign):
     :returns: Query parameters matching the signing credentials with a
               signed payload.
     """
-    if not hasattr(credentials, 'sign_blob'):
+    if not isinstance(credentials, google.auth.credentials.Signing):
```

Comment:
This is the first (of many) time in recorded history that your mixins were useful